datasets/msrvid.py

```diff
@@ -0,0 +1,94 @@
+import math
+import os
+
+import numpy as np
+import torch
+from torchtext.data.dataset import Dataset
+from torchtext.data.example import Example
+from torchtext.data.field import Field
+from torchtext.data.iterator import BucketIterator
+from torchtext.data.pipeline import Pipeline
+from torchtext.vocab import Vectors
+
+from datasets.idf_utils import get_pairwise_word_to_doc_freq, get_pairwise_overlap_features
+
+
+def get_class_probs(sim, *args):
+    """
+    Convert a single label into class probabilities.
+    """
+    class_probs = np.zeros(MSRVID.NUM_CLASSES)
+    ceil, floor = math.ceil(sim), math.floor(sim)
+    if ceil == floor:
+        class_probs[floor] = 1
+    else:
+        class_probs[floor] = ceil - sim
+        class_probs[ceil] = sim - floor
+
+    return class_probs
+
+
+class MSRVID(Dataset):
+    NAME = 'msrvid'
+    NUM_CLASSES = 6
+    ID_FIELD = Field(sequential=False, use_vocab=False, batch_first=True)
+    TEXT_FIELD = Field(batch_first=True, tokenize=lambda x: x)  # tokenizer is identity since we already tokenized it to compute external features
```

Comment:
Would a better way be `sequential = False`?

Reply:
Hmm, a text field (a series of words) is sequential though - so not sure if that will be confusing so I used an identity tokenize function.